### ISSUE TITLE
[feature] gitpod support

### ISSUE THREAD

**Issue by keesj-exset** [d-koppenhagen/vscode-code-review#95]
# 🧩 Feature

It would be nice if is was possible to run this extension in gitpod to facilitate review. Git pod is an open source cloud based editor based on I think the same code based as vs code.(  https://www.gitpod.io/docs/ide/ ).

## Description

It would be nice if is was possible to run this extension in gitpod to facilitate review. Git pod is an open source cloud based editor based on I think the same code based as vs code.(  https://www.gitpod.io/docs/ide/ ).

Anyway the extension is not available on their "extension store" https://www.gitpod.io/blog/open-vsx/ so I tried
packaging it myself and "drag and dropping" it into gitpod. https://www.gitpod.io/blog/open-vsx-in-gitpod/

Packaging was done doing

npm install
vsce package

And drag and dropping the file (vscode-code-review-1.23.0.vsix in this case)
but this did not directly work. the code-review.csv file gets created and some menu options are visible but the add notes remains empty.

![empty](https://user-images.githubusercontent.com/62298192/106471439-e934dc80-64a1-11eb-87f4-a5a99110c5ad.png)

for short.. there are at least two problems:
* The extension that I compiled/install does not work on gitpod.
* The extension is not "published" on the alternative repository.

**Comment by d-koppenhagen** [comment]
Hey @keesj-exset thanks for this feature suggestion. I just published the [Current version in the open VSX registry](https://open-vsx.org/extension/d-koppenhagen/vscode-code-review).

You should now be able to find this extension directly in the Gitpods extension panel and install it.

I just tried it our and it seems to work as expected.

Please double check this and give me feedback if everything works for you. If yes: we can close this issue.

Thanks again for the idea!

**Comment by michelcaradec** [comment]
Hello @keesj-exset , the problem seems to come from Firefox and its ServiceWorker management.
When adding a note, and displaying the form, the following errors show in the console:

```raw
Failed to register/update a ServiceWorker for scope ‘https://webview-${SOME_GUID}.ws-eu03.gitpod.io/webview/’: Storage access is restricted in this context due to user settings or private browsing mode.
Uncaught (in promise) DOMException: The operation is insecure.
```

This post gives some explanation, and some workaround: <https://dev.webonomic.nl/fixing-firefox-failed-to-register-update-a-serviceworker-for-scope>.
I applied it and now it works.

**Comment by keesj-exset** [comment]
> Hello @keesj-exset , the problem seems to come from Firefox and its ServiceWorker management.

So nice of you to .. find this problem and point it out!

**Comment by keesj-exset** [comment]
@d-koppenhagen I was able to install the extension by searching in the store and have seen it fully working when using chrome. This is really a nice experience.

I also tried to enable the firefox workarounds but this did not work. For now I am quite happy but I would like to pursue this issue is this issue cause by this plugin, gitpod or me? With kind regards

![image10](https://user-images.githubusercontent.com/62298192/106566595-c998c500-6530-11eb-9a60-a845fa3128a0.png)

**Comment by d-koppenhagen** [comment]
@michelcaradec correct me if I am wrong but I think it's a VSCode API / GitPod issue.
In the Plugin source code, Service Workers aren't used by intention. I think they will be used underneath some API layer from the VSCode extension API.

**Comment by michelcaradec** [comment]
> @michelcaradec correct me if I am wrong but I think it's a VSCode API / GitPod issue.
> In the Plugin source code, Service Workers aren't used by intention. I think they will be used underneath some API layer from the VSCode extension API.

Hello @d-koppenhagen ,
I'm not an expert on this one, but after a bot of investigation, I found [this VS Code issue](https://github.com/microsoft/vscode/issues/87076), mentioning there is no immediate plan to support Service Worker (end of december 2019).

I also found this recent [Stack Overflow post](https://stackoverflow.com/questions/65999175/do-vscode-webviews-support-web-workers) reporting a similar issue.

So I'm afraid that we will have to live with it.

By the way @keesj-exset , the option to disable in Firefox can be found in `about:preferences#privacy`:

![2021-02-03_08-30](https://user-images.githubusercontent.com/12159769/106713078-34620300-65fa-11eb-90f6-951d9413237a.png)
